### PR TITLE
feat(css/parser): `@layer` at-rule

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

`@layer` parsing https://www.w3.org/TR/css-cascade-5/#at-layer

There are couple problems were found:
- `@import` is not supported `supports()`:
```css
@import url("narrow.css") supports(display: flex) handheld and (max-width: 400px);
```
- `@supports` is not supported `declaration` parsing and `selector` parsing

These things will be fixed in future

**BREAKING CHANGE:**

Do we have it for CSS?

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_codegen/src/list.rs [hunk 1]
```diff
@@ -29,7 +29,8 @@ add_bitflags!(
         SpaceDelimited: 1 << 2,
         SemiDelimited: 1 << 3,
         CommaDelimited: 1 << 4,
-        DelimitersMask: SpaceDelimited | SemiDelimited | CommaDelimited,
+        DotDelimited: 1 << 5,
+        DelimitersMask: SpaceDelimited | SemiDelimited | CommaDelimited | DotDelimited,
```
Comment (kdy1) [inline crates/swc_css_codegen/src/list.rs:33]: I think we should change value of`AllowTrailingComma` to `1 << 6`>

**Review by kdy1 — CHANGES_REQUESTED** [review]
Mostly LGTM, thanks!

I found an issue related to bitflags.

**Comment by alexander-akait** [comment]
@kdy1 Fixed :+1:

**Review by kdy1 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**